src/source-pcap.c

```diff
@@ -129,14 +149,52 @@ void TmModuleDecodePcapRegister (void)
     tmm_modules[TMM_DECODEPCAP].flags = TM_FLAG_DECODE_TM;
 }
 
+/**
+ * \brief Update 64 bit |last| value from |current32| value taking one
+ * wrap-around into account.
+ */
+static inline void UpdatePcapStatsValue64(uint64_t *last, uint32_t current32)
+{
+    /* uint64_t -> uint32_t is defined behaviour. It slices lower 32bits. */
+    uint32_t last32 = *last;
```

Comment:
@victorjulien I added a comment about "being defined behaviour". As mentioned in the previous review comment, this is defined behaviour as it's unsigned -> unsigned (smaller).

Does that help? Or would you like me to add masking and casting?

```
    // this is the same
    uint32_t last32 = (uint32_t)(*last & (uint64_t)UINT32_MAX); // or 0xFFFFFFFFull instead of UINT32_MAX
```

Reply:
No, please keep it as is. The comment is helpful for ppl like me who depend on ubsan to know what undefined behavior is :)